File: geometrikortalama.py
dizi=[]
for i in range(1000):

    sayi=input("Sayıyı giriniz.(İşlem bitince boş enter gönderiniz):")

    if(sayi==""):
        break
    dizi.append(sayi)

def geometrik(dizi):
    carpim=1
    for a in range(len(dizi)):
        carpim=carpim*int((dizi[a]))

    n=len(dizi)

    return (round(carpim**(1/n),4))

print("Girdiğiniz dizinin geometrik ortalaması:")
print(geometrik(dizi))

File: GeometrikOrtalama.java
public class GeometrikOrtalama 
{
    public static void main(String[] args) 
    {
        int [ ] dizi = {1,2,3,6};
        double  carpim = 1;
        
        for (int i = 0; i < dizi.length; i++) 
        {
            carpim *= dizi[i]; 
        }
        
        System.out.println("Dizideki elemnaların geometrik ortalaması = " + Math.sqrt(carpim));
    }   
}
